src/Strategy/Archive/RPM.hs

```diff
@@ -0,0 +1,41 @@
+module Strategy.Archive.RPM
+  ( extractRpm
+  )
+where
+
+import qualified Codec.RPM.Conduit as RPM
+import Conduit
+import Control.Effect.Lift
+import Control.Exception (throwIO)
+import Control.Monad.Except
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Char8 as BS8
+import qualified Data.ByteString.Lazy as BL
+import qualified Data.CPIO as CPIO
+import Path
+import Prelude
+import qualified Path.IO as PIO
+
+extractRpm :: Has (Lift IO) sig m => Path Abs Dir -> Path Abs File -> m ()
+extractRpm dir rpmFile = do
+  res <-
+    sendIO . runResourceT . runExceptT . runConduit $
+      readRPMEntries rpmFile .| filterC (not . CPIO.isEntryDirectory) .| sinkDir
```

Comment:
This uses a streaming library called `conduit` -- you can think of the `.|` operators as "unix pipes".

As for how to read the conduit type signatures: `ConduitT i o m a` has:

- "streaming input type" `i`
- "streaming output type" `o`
- underlying monad `m` (it's a monad transformer)
- result type `a` (like any other monad)